scipy/stats/_mstats_basic.py

```diff
@@ -348,6 +348,8 @@ def _mode1D(a):
             slices[axis] = 1
             counts = output[tuple(slices)].reshape(newshape)
             output = (modes, counts)
+        else:
+            output = np.moveaxis(output, axis, 0)
```

Comment:
The problem was that the following statement, `ModeResult(*output)`, assumed that the (conceptually) new axis corresponding with the separate mode/count results would always be in the 0th axis. That is incorrect; when `np.apply_along_axis` argument `func1d` returns an array, it adds this new axis in the place of the `axis` it reduced. This moves that axis to the front as the `ModeResult(*output)` statement expects.